modules/core/src/test/java/org/apache/ignite/util/GridCommandHandlerTest.java

```diff
@@ -2599,29 +2626,58 @@ public void testDiagnosticPageLocksTracker() throws Exception {
 
         ignite.cluster().active(true);
 
-        String dir = U.defaultWorkDirectory() + "/diagnostic/";
-
         assertEquals(EXIT_CODE_OK, execute("--diagnostic"));
         assertEquals(EXIT_CODE_OK, execute("--diagnostic", "help"));
-        assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "help"));
 
         assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump"));
+
+        checkNumberFiles(defaultDiagnosticDir, 1);
+
         assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump_log"));
-        assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump", dir));
+        assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump", "--path",
+            customDiagnosticDir.getAbsolutePath()));
+
+        checkNumberFiles(customDiagnosticDir, 1);
 
         assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump", "--all"));
+
+        checkNumberFiles(defaultDiagnosticDir, 5);
+
         assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump_log", "--all"));
-        assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump", dir, "--all"));
+        assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump",
+            "--path", customDiagnosticDir.getAbsolutePath(), "--all"));
+
+        checkNumberFiles(customDiagnosticDir, 5);
+
+        assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump",
+            "--nodes", node0.id().toString() + "," + node2.id().toString()));
```

Comment:
Your could use for CommandLogger#join() method for joining objects with specified delimeter